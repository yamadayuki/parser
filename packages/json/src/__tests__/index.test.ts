import { parse } from "..";

describe("parse", () => {
  test("returns the result of parsing json string", () => {
    const result = parse('{ "foo": 1 }');
    expect(result.lexerErrors).toEqual([]);
    expect(result.parserErrors).toEqual([]);
    expect(result.json).toMatchSnapshot();
  });
});
