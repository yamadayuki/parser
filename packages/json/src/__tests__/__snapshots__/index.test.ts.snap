// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parse returns the result of parsing json string 1`] = `
Object {
  "children": Object {
    "object": Array [
      Object {
        "children": Object {
          "LCurly": Array [
            Object {
              "endColumn": 1,
              "endLine": 1,
              "endOffset": 0,
              "image": "{",
              "startColumn": 1,
              "startLine": 1,
              "startOffset": 0,
              "tokenType": Object {
                "CATEGORIES": Array [],
                "PATTERN": "{",
                "categoryMatches": Array [],
                "categoryMatchesMap": Object {},
                "isParent": false,
                "name": "LCurly",
                "tokenTypeIdx": 6,
              },
              "tokenTypeIdx": 6,
            },
          ],
          "RCurly": Array [
            Object {
              "endColumn": 12,
              "endLine": 1,
              "endOffset": 11,
              "image": "}",
              "startColumn": 12,
              "startLine": 1,
              "startOffset": 11,
              "tokenType": Object {
                "CATEGORIES": Array [],
                "PATTERN": "}",
                "categoryMatches": Array [],
                "categoryMatchesMap": Object {},
                "isParent": false,
                "name": "RCurly",
                "tokenTypeIdx": 7,
              },
              "tokenTypeIdx": 7,
            },
          ],
          "objectItem": Array [
            Object {
              "children": Object {
                "Colon": Array [
                  Object {
                    "endColumn": 8,
                    "endLine": 1,
                    "endOffset": 7,
                    "image": ":",
                    "startColumn": 8,
                    "startLine": 1,
                    "startOffset": 7,
                    "tokenType": Object {
                      "CATEGORIES": Array [],
                      "PATTERN": ":",
                      "categoryMatches": Array [],
                      "categoryMatchesMap": Object {},
                      "isParent": false,
                      "name": "Colon",
                      "tokenTypeIdx": 11,
                    },
                    "tokenTypeIdx": 11,
                  },
                ],
                "StringLiteral": Array [
                  Object {
                    "endColumn": 7,
                    "endLine": 1,
                    "endOffset": 6,
                    "image": "\\"foo\\"",
                    "startColumn": 3,
                    "startLine": 1,
                    "startOffset": 2,
                    "tokenType": Object {
                      "CATEGORIES": Array [],
                      "PATTERN": /"\\(\\?:\\[\\^\\\\\\\\"\\]\\|\\\\\\\\\\(\\?:\\[bfnrtv"\\\\\\\\\\\\/\\]\\|u\\[0-9a-fA-F\\]\\{4\\}\\)\\)\\*"/,
                      "categoryMatches": Array [],
                      "categoryMatchesMap": Object {},
                      "isParent": false,
                      "name": "StringLiteral",
                      "tokenTypeIdx": 12,
                    },
                    "tokenTypeIdx": 12,
                  },
                ],
                "value": Array [
                  Object {
                    "children": Object {
                      "NumberLiteral": Array [
                        Object {
                          "endColumn": 10,
                          "endLine": 1,
                          "endOffset": 9,
                          "image": "1",
                          "startColumn": 10,
                          "startLine": 1,
                          "startOffset": 9,
                          "tokenType": Object {
                            "CATEGORIES": Array [],
                            "PATTERN": /-\\?\\(0\\|\\[1-9\\]\\\\d\\*\\)\\(\\\\\\.\\\\d\\+\\)\\?\\(\\[eE\\]\\[\\+-\\]\\?\\\\d\\+\\)\\?/,
                            "categoryMatches": Array [],
                            "categoryMatchesMap": Object {},
                            "isParent": false,
                            "name": "NumberLiteral",
                            "tokenTypeIdx": 13,
                          },
                          "tokenTypeIdx": 13,
                        },
                      ],
                    },
                    "name": "value",
                  },
                ],
              },
              "name": "objectItem",
            },
          ],
        },
        "name": "object",
      },
    ],
  },
  "name": "json",
}
`;
